### PR TITLE
🎨  Improved Update Notifications

### PR DESCRIPTION
**Pull request by kirrg001**
refs #5071 

We want to achieve these things:

1. Get rid of the hardcoded notification message - receive messages from the update check service.
2. Be able to receive custom messages.
3. Be less annoying and turn `dismissible` on by default.
4. Future proof dynamic (e.g. be able to receive multiple notifications, notification groups)
5. Custom Update Check endpoint for development.
6. We stick with top bar notification in LTS, because we only release LTS once per 1-3 month and with this PR we are able to send custom messages, but for 1.0 we will move default release notifications into the about page. And urgent/priority notifications will show up on the top bar e.g. security updates, migration notifications etc.

**Note**: I deleted the `displayUpdateNotification` settings key, because it's not used anymore. No need to add a migration still, as LTS ends anyway this year. 

**Note**: I've renamed `seenNotifications` to `notifications`. `notifications` is the true place to store/read notifications from. Unique identifier is a uuid. We simply store any new notification in it. If you remove a notification, it get's marked as seen. And deleted after 3 month automatically. No need to add a migration still, as LTS ends anyway this year - but we'll do for 1.x.

**Note**: No usage of notification in process store anymore, because this now that the messages are coming from the UNC, we can't simply store the messages in process. Because if you receive a notification and you restart Ghost, the message is lost. You would get it back on the next request to the service. No need to cache the notifications in general, because it's only one read operation to one key in the settings table.

**Note** Each commit message has all the content you need.

- [x] remove `config.updateCheck`, keep the privacy flag and remove this flag from Pro
- [x] add possibility to receive multiple release notifications (just in case we want that in the future)
- [x] fix tests / add tests
- [x] change privacy flag behaviour
- [x] care about restart use case (therefor we have to store notifications in the settings table)
- [x] add condition to receive custom pro notification for migrations only if blog is allowed to migrate -> generalise this by using e.g. config.allowedGroupMessage (an array of regex strings e.g. `migration` or `share-more`.
- [x] future case: be able to send a custom notification to a single blog on pro
- [x] code quality check/read commit messages again/overview of the last changes

### CODE REVIEW

File: core/server/data/schema/default-settings.json [hunk 1]
```diff
@@ -9,9 +9,6 @@
         "nextUpdateCheck": {
             "defaultValue": null
         },
-        "displayUpdateNotification": {
-            "defaultValue": null
-        },
```
Comment (ErisDS) [inline core/server/data/schema/default-settings.json]: How are we keeping track of notifications in this updated logic?

**Comment by ErisDS** [comment]
I tried to add in a client extension for showing an extra notification from config. That got complicated quickly, and developing on LTS is so crazy hard now!

I would suggest that somewhere in this server side code where the notifications are fetched, it would be great to merge in `config.notifications` if it is set and an array. That would solve our one last case. I've tried to do this, but it ends up touching the same areas as you're already working on.

**Comment by kirrg001** [comment]
> would suggest that somewhere in this server side code where the notifications are fetched, it would be great to merge in config.notifications if it is set and is an array. That would solve our one last case. I've tried to do this, but it ends up touching the same areas as you're already working on.

Will check tomorrow.

**Comment by kirrg001** [comment]
Ready to review!

**Pull request merged**